doc/chips/9.rst

```diff
@@ -292,6 +304,49 @@ the user's home directory by default:
     (dependencies)
 
 
+Syncing Commands
+++++++++++++++++
+
+When any of ``add``, ``rm``, or ``update`` are invoked, ``mason`` will make sure that
+the manifest file, lock file, and dependency code are kept in sync with each other.
+Effectively, this means that when ``update`` is run, dependencies will be downloaded,
+and when ``add`` or ``rm`` are run, ``update`` will be run.
+
+To avoid unnecessary updates, a hash of the manifest will be kept in the lock file.
+We will not hash the lock file because the dependencies across all projects are kept
+in a single namespaced directory -- we can simply check if all the dependency
+directories are present. 
+
+
+Build System
+------------
+
+When invoked, ``mason build`` will do the following:
+
+* Navigate to the root of the project.
+
+* Run ``update`` to make sure any manual manifest edits are reflected in the
+  dependency code.
+
+* Build ``main.chpl`` in the ``/src`` directory.
```

Comment:
This naming convention of `main.chpl` may be somewhere we'd like to deviate from Cargo, given the equivalence of Chapel file names and module names (if a module is not specified within).